osu.Game/Graphics/UserInterface/BackButton.cs

```diff
@@ -0,0 +1,165 @@
+                        {
+                            Anchor = Anchor.Centre,
+                            TextSize = 25,
+                            Icon = FontAwesome.fa_osu_left_o
+                        },
+                    }
+                },
+                rightContainer = new Container
+                {
+                    Origin = Anchor.TopLeft,
+                    Anchor = Anchor.TopLeft,
+                    RelativeSizeAxes = Axes.Y,
+                    Width = 80,
+                    Position = Position + new Vector2(initial_extend_length, 0),
+                    Children = new Drawable[]
+                    {
+                        new Box
+                        {
+                            Colour = new Color4(238, 51, 153, 255),
+                            Origin = Anchor.TopLeft,
+                            Anchor = Anchor.TopLeft,
+                            RelativeSizeAxes = Axes.Both,
+                            Shear = new Vector2(shear, 0),
+                            EdgeSmoothness = new Vector2(1.5f, 0),
+                        },
+                        new SpriteText
+                        {
+                            Origin = Anchor.Centre,
+                            Anchor = Anchor.Centre,
+                            Text = @"Back",
+                        }
+                    }
+                }
+            };
+        }
+        protected override bool OnHover(InputState state)
+        {
+            icon.ClearTransformations();
+
+            rightContainer.MoveToX(Position.X + extend_length, transform_time, EasingTypes.OutElastic);
+            leftContainer.ResizeTo(new Vector2(extend_length, 1.0f), transform_time, EasingTypes.OutElastic);
+
+            Width = width_extended; // right container + ExtendLength
+
+            int duration = 0; //(int)(Game.Audio.BeatLength / 2);
+            if (duration == 0) duration = pulse_length;
+
+            double offset = 0; //(1 - Game.Audio.SyncBeatProgress) * duration;
+            double startTime = Time.Current + offset;
+
+            // basic pulse
+            icon.Transforms.Add(new TransformScale
+            {
+                StartValue = new Vector2(1.1f),
+                EndValue = Vector2.One,
+                StartTime = startTime,
+                EndTime = startTime + duration,
+                Easing = EasingTypes.Out,
+                LoopCount = -1,
+                LoopDelay = duration
+            });
+
+            return true;
+        }
+
+        protected override void OnHoverLost(InputState state)
+        {
+            icon.ClearTransformations();
+
+            rightContainer.MoveToX(Position.X + initial_extend_length, transform_time, EasingTypes.OutElastic);
+            leftContainer.ResizeTo(new Vector2(initial_extend_length, 1.0f), transform_time, EasingTypes.OutElastic);
+
+            Width = width_retracted; // right container + InitialExtendLength

```

Comment:
so i should anchor right container to right side of the button and it should resize only whole button and left container, right?